CoreFoundation/Base.subproj/CFBase.h

```diff
@@ -144,6 +144,17 @@
 
 #endif
 
+#if DEPLOYMENT_TARGET_WINDOWS
+  // No C99 support
+  #define CF_RESTRICT
+#else
+  #if defined(__cplusplus)
+    #define CF_RESTRICT __restrict__
+  #else
+    #define CF_RESTRICT restrict
+  #endif
+#endif
+
```

Comment:
This is a public header so we can't just add the macro here. Perhaps `ForSwiftFoundationOnly.h`?

Reply:
The `restrict` is used in CFPlatform.c and CFStorage.c, so could move it into CFInternal.h instead?

Reply:
I was thinking CFInternal as well, but then it's used in ForSwiftFoundationOnly.h. I'd prefer the internal header if possible.

Reply:
So it looks like if we put it in `CFInternal.h` then it can't be used for the ForSwiftFoundationOnly.h. We could have the macro in CFInternal for ease of maintenance and then have a special-case for the ForSwiftFoundationOnly that duplicates it, but it's not elegant.

Reply:
So @parkera I have moved the definitions to be internal only, but can't use it in the ForSwiftFoundation file. I'm not sure which approach you'd like me to take here in terms of conditionally defining the value of `restrict` -- I've duplicated the line here, but it's introducing duplicate code which I don't like. Any other preferences/suggestions?